src/Textile/API/events.ts

```diff
@@ -0,0 +1,279 @@
+  )
+}
+
+// tslint:disable-next-line:max-line-length
+export function addNotificationReceivedListener(listener: (notification: INotification) => void) {
+  notificationReceivedListeners.push(listener)
+  return new EventSubscription(
+    () => notificationReceivedListeners = notificationReceivedListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addThreadUpdateReceivedListener(listener: (feedItem: IFeedItem) => void) {
+  threadUpdateReceivedListeners.push(listener)
+  return new EventSubscription(
+    () => threadUpdateReceivedListeners = threadUpdateReceivedListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addThreadAddedListener(listener: (threadId: string) => void) {
+  threadAddedListeners.push(listener)
+  return new EventSubscription(
+    () => threadAddedListeners = threadAddedListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addThreadRemovedListener(listener: (threadId: string) => void) {
+  threadRemovedListeners.push(listener)
+  return new EventSubscription(
+    () => threadRemovedListeners = threadRemovedListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addAccountPeerAddedListener(listener: (peerId: string) => void) {
+  accountPeerAddedListeners.push(listener)
+  return new EventSubscription(
+    () => accountPeerAddedListeners = accountPeerAddedListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addAccountPeerRemovedListener(listener: (peerId: string) => void) {
+  accountPeerRemovedListeners.push(listener)
+  return new EventSubscription(
+    () => accountPeerRemovedListeners = accountPeerRemovedListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addQueryDoneListener(listener: (queryId: string) => void) {
+  queryDoneListeners.push(listener)
+  return new EventSubscription(
+    () => queryDoneListeners = queryDoneListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addQueryErrorListeners(listener: (queryId: string, error: string) => void) {
+  queryErrorListeners.push(listener)
+  return new EventSubscription(
+    () => queryErrorListeners = queryErrorListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addClientThreadQueryResultListeners(listener: (queryId: string, thread: IThread) => void) {
+  clientThreadQueryResultListeners.push(listener)
+  return new EventSubscription(
+    () => clientThreadQueryResultListeners = clientThreadQueryResultListeners.filter((item) => item !== listener),
+  )
+}
+
+export function addContactQueryResultListeners(listener: (queryId: string, contact: IContact) => void) {
+  contactQueryResultListeners.push(listener)
+  return new EventSubscription(
+    () => contactQueryResultListeners = contactQueryResultListeners.filter((item) => item !== listener),
+  )
+}
+
+eventEmitter.addListener('NODE_STARTED', () => {
```

Comment:
Here is where the sdk subscribes to all the raw untyped events coming from the RN bridge.